packages/data-service/src/data-service.ts

```diff
@@ -713,6 +714,13 @@ export interface DataService {
    * Register reauthentication handlers with this DataService instance.
    */
   addReauthenticationHandler(handler: ReauthenticationHandler): void;
+
+  /**
+   * Return the current state of ConnectionOptions secrets, which may have changed
+   * since connecting (e.g. OIDC tokens). The `connectionInfoSecretsChanged` event
+   * is being emitted when this value changes.
+   */
+  getUpdatedSecrets(): Promise<Partial<ConnectionOptions>>;
```

Comment:
Just a random thought, somehow I would rather have something less generic so the domain and intent stands out more, both here and in the `connectionMergeInfos`. Like  `getUpdatedOIDCTokens`. In know is less future-proof but we could easily refactor it later if we would have more secrets.